src/module.ts

```diff
@@ -161,43 +126,15 @@ export default defineNuxtModule<ModuleOptions>({
           ...(typeof editorSupportConfig.autocompleteUtil === 'object' ? editorSupportConfig.autocompleteUtil : {})
         })
       }
-
-      if (editorSupportConfig.generateConfig || isNuxt2()) {
-        addTemplate({
-          filename: 'tailwind.config.cjs',
-          getContents: () => `module.exports = ${JSON.stringify(resolvedConfig, null, 2)}`,
-          write: true,
-          ...(typeof editorSupportConfig.generateConfig === 'object' ? editorSupportConfig.generateConfig : {})
-        })
-      }
     }
 
     // enabled only in development
     if (nuxt.options.dev) {
-      // Watch the Tailwind config file to restart the server
-      if (isNuxt2()) {
-        nuxt.options.watch = nuxt.options.watch || []
-        configPaths.forEach(path => nuxt.options.watch.push(path))
-      } else if (Array.isArray(nuxt.options.watch)) {
-        configPaths.forEach(path => nuxt.options.watch.push(relative(nuxt.options.srcDir, path)))
-      } else {
-        const watcher = watch(configPaths, { depth: 0 }).on('change', (path) => {
-          logger.info(`Tailwind config changed: ${path}`)
-          logger.warn('Please restart the Nuxt server to apply changes or upgrade to latest Nuxt for automatic restart.')
-        })
-        nuxt.hook('close', () => watcher.close())
-      }
-
-      // Insert Vite plugin to work around HMR issue
-      if (!moduleOptions.disableHmrHotfix) {
-        addVitePlugin(vitePlugin(tailwindConfig, nuxt.options.rootDir, resolvedCss))
-      }
-
       // Add _tailwind config viewer endpoint
       // TODO: Fix `addServerHandler` on Nuxt 2 w/o Bridge
       if (moduleOptions.viewer) {
-        const viewerConfig = resolveViewerConfig(moduleOptions.viewer)
-        setupViewer(tailwindConfig, viewerConfig, nuxt)
+        const viewerConfig = resolvers.resolveViewerConfig(moduleOptions.viewer)
+        setupViewer(twConfig.tailwindConfig, viewerConfig, nuxt)
```

Comment:
Pass twConfig as-is, and use `tailwindcss/loadConfig` in tailwindConfigProvider function.